client/search-ui/src/input/SearchHistoryDropdown.tsx

```diff
@@ -0,0 +1,179 @@
+            </>
+        )
+    }
+)
+
+interface SearchHistoryEntriesProps {
+    recentSearches: RecentSearch[]
+    onSelect: (search: RecentSearch) => void
+}
+
+const SearchHistoryEntries: React.FunctionComponent<SearchHistoryEntriesProps> = ({ recentSearches, onSelect }) => {
+    const { isOpen } = usePopoverContext()
+    const [selectedIndex, setSelectedIndex] = useState(0)
+    const selectedIndexRef = useRef(selectedIndex)
+    selectedIndexRef.current = selectedIndex
+
+    const keydownHandler: KeyboardEventHandler<HTMLElement> = useCallback(
+        (event: KeyboardEvent) => {
+            switch (event.key) {
+                case 'ArrowDown':
+                    event.preventDefault()
+                    setSelectedIndex(index => (index + 1) % recentSearches.length)
+                    break
+                case 'ArrowUp':
+                    event.preventDefault()
+                    setSelectedIndex(index => (index - 1 + recentSearches.length) % recentSearches.length)
+                    break
+                case 'Enter':
+                    event.preventDefault()
+                    if (recentSearches.length > 0) {
+                        onSelect(recentSearches[selectedIndexRef.current])
+                    }
+                    break
+            }
+        },
+        [setSelectedIndex, recentSearches, onSelect]
+    )
+
+    const clickHandler: MouseEventHandler<HTMLElement> = useCallback(
+        (event: MouseEvent<HTMLElement>) => {
+            if (event.button === 0) {
+                event.preventDefault()
+                const index = (event.target as HTMLElement).closest('li')?.dataset.index
+                if (index !== undefined) {
+                    onSelect(recentSearches[+index])
+                }
+            }
+        },
+        [recentSearches, onSelect]
+    )
+
+    if (!isOpen) {
+        return null
+    }
+
+    if (recentSearches.length === 0) {
+        return <div className="text-muted px-3 py-2">Your recent searches will appear here</div>
+    }
+
+    return (
+        <ul
+            className={styles.list}
+            ref={ref => ref?.focus()}
+            aria-expanded="true"
+            role="listbox"
+            tabIndex={0}
+            onKeyDown={keydownHandler}
+            onClick={clickHandler}
+        >
+            {recentSearches.map((search, index) => (
+                <SearchHistoryEntry key={index} index={index} search={search} selected={index === selectedIndex} />
+            ))}
+        </ul>
+    )
+}
+
+const SearchHistoryEntry: React.FunctionComponent<{
+    index: number
+    search: RecentSearch
+    selected: boolean
+}> = React.memo(({ index, search, selected }) => (
+    <li role="option" data-index={index} aria-selected={selected}>
+        <SyntaxHighlightedSearchQuery query={search.query} tabIndex={-1} />
+        <span className="ml-1 text-nowrap text-muted">
+            <Timestamp date={search.timestamp} />
```

Comment:
We need to add a hidden pause here of some kind to stop the screen reader from smooshing the date with the query.
```suggestion
            <span class="sr-only">,</span>
            <Timestamp date={search.timestamp} />
```